### PR TITLE
Chore: Let kbn util infer types

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

This pr moves all functions and properties inside the kbn util object. This way we will have type safety.

There were two type related errors that I fixed. One in Google Cloud computing which I've checked that it gives back a number for sure.
Stackdriver - instance_disk_throttled__ - Grafana - Google Chrome 2020-08-10 20_24_03

The other was in Graphite where I created a condition.

### CODE REVIEW

File: public/app/core/utils/kbn.ts [hunk 1]
```diff
@@ -10,310 +10,285 @@ import {
...
+    y: 31536000,
+    M: 2592000,
+    w: 604800,
+    d: 86400,
+    h: 3600,
+    m: 60,
+    s: 1,
+    ms: 0.001,
+  } as { [index: string]: number },
+  regexEscape: (value: string) => {
```
Comment (kaydelaney) [inline public/app/core/utils/kbn.ts:26]: Could be simplified further with
```typescript
regexEscape: (value: string) =>  value.replace(/[\\^$*+?.()|[\]{}\/]/g, '\\$&'),
```

File: public/app/plugins/datasource/graphite/MetricTankMetaInspector.tsx [hunk 2]
```diff
@@ -25,7 +25,7 @@ export class MetricTankMetaInspector extends PureComponent<Props, State> {
     let totalSeconds = 0;
 
     for (const bucket of buckets) {
-      totalSeconds += kbn.interval_to_seconds(bucket.retention);
+      totalSeconds += bucket.retention ? kbn.interval_to_seconds(bucket.retention) : 0;
```
Comment (kaydelaney) [inline public/app/plugins/datasource/graphite/MetricTankMetaInspector.tsx:28]: Could rewrite this using reduce (though this is personal preference of course)

```typescript
const totalSeconds = buckets.reduce((acc, bucket) => acc + (bucket.retention ? kbn.interval_to_seconds(bucket.retention) : 0));
```

File: public/app/core/utils/kbn.ts [hunk 3]
```diff
@@ -10,310 +10,285 @@ import {
...
+    const strings: string[] = [];
+    const numHours = Math.floor(seconds / 3600);
+    const numMinutes = Math.floor((seconds % 3600) / 60);
+    const numSeconds = Math.floor((seconds % 3600) % 60);
+    numHours > 9 ? strings.push('' + numHours) : strings.push('0' + numHours);
+    numMinutes > 9 ? strings.push('' + numMinutes) : strings.push('0' + numMinutes);
+    numSeconds > 9 ? strings.push('' + numSeconds) : strings.push('0' + numSeconds);
+    return strings.join(':');
+  },
+  to_percent: (nr: number, outOf: number) => {
```
Comment (kaydelaney) [inline public/app/core/utils/kbn.ts:157]: Could be simplified further with
```typescript
to_percent: (nr: number, outOf: number) => `${Math.floor((nr / outOf) * 10000) / 100}%`
```

**Review by kaydelaney — COMMENTED** [review]
Some small comments, otherwise looks good

**Review by torkelo — COMMENTED** [review]
Awesome! :heart: love PRs like this, been wanting to refactor kbn for years :)

File: public/app/plugins/datasource/graphite/MetricTankMetaInspector.tsx (hunk 2 above)
Comment (zoltanbedi) [reply public/app/plugins/datasource/graphite/MetricTankMetaInspector.tsx:28]: Neat, I like that solution.

File: public/app/core/utils/kbn.ts [hunk 4]
```diff
@@ -1,319 +1,291 @@
...
-  s: 1,
-  ms: 0.001,
-};
-
-kbn.calculateInterval = (range: TimeRange, resolution: number, lowLimitInterval: string[]) => {
-  let lowLimitMs = 1; // 1 millisecond default low limit
-  let intervalMs;
+const kbn = {
+  valueFormats: {} as ValueFormatterIndex,
+  interval_regex: /(\d+(?:\.\d+)?)(ms|[Mwdhmsy])/,
```
Comment (dprokop) [inline public/app/core/utils/kbn.ts:16]: Since you are on it, maybe we could unify the naming here to be pascalCase?

Comment (zoltanbedi) [reply public/app/core/utils/kbn.ts:16]: sure, I'll do that.

**Review by kaydelaney — APPROVED** [review]
LGTM

**Review by dprokop — APPROVED** [review]

**Pull request merged**